abis_mapping/plugins/wkt.py

```diff
@@ -0,0 +1,105 @@
+"""Provides custom frictionless wkt plugin for the package."""
+
+
+# Third-party
+import frictionless
+import shapely
+
+# Typing
+from typing import Optional, Any, Type
+
+
+class WKTPlugin(frictionless.Plugin):
+    """Custom WKT plugin."""
+
+    # Class attributes
+    code = "wkt"
+
+    def select_field_class(
+        self,
+        type: Optional[str] = None
+    ) -> Optional[Type[frictionless.Field]]:
+        """Return the custom field class for the plugin.
+
+        Args:
+            type (str): Denotes the type passed int from the frictionless
+                framework for a field.
+
+        Returns:
+            Optional[Type[frictionless.Field]]: Reference to WKTField if type
+                matches code else None.
+        """
+        if type == self.code:
+            return WKTField
+        return None
+
+
+class WKTField(frictionless.Field):
+    """Custom WKT Type implementation."""
+
+    # Class attributes
+    type = "wkt"
+    builtin = False
+    supported_constraints = [
+        "required",
+    ]
+
+    def create_value_reader(self) -> frictionless.schema.types.IValueReader:
+        """Creates value reader callable."""
+
+        def value_reader(cell: Any) -> Optional[shapely.Geometry]:
+            """Convert cell (read direction).
+
+            Args:
+                cell (Any): Cell to convert
+
+            Returns:
+                Optional[shapely.Geometry]: Converted cell or None if invalid
+            """
+            # Expect the cell to be a string.
+            if not isinstance(cell, str):
+                return None
+
+            # Attempt to convert the WKT string
+            try:
+                geometry = shapely.from_wkt(cell)
+            except shapely.GEOSException:
+                return None
+
+            # Return the CRS object
+            return geometry
+
+        # Return value_reader callable
+        return value_reader
+
+    def create_value_writer(self) -> frictionless.schema.types.IValueWriter:
+        """Creates value write callable."""
+
+        def value_writer(cell: shapely.Geometry) -> str:
+            """Convert cell (write direction).
+
+            Args:
+                cell (pyproj.CRS): Cell to convert.
+
+            Returns:
+                str: Converted cell.
+
+            Raises:
+                TypeError: if the returned value from the shapely
+                    method to_wkt is not string.
+            """
+            # Serialize to default format for pyproj
+            wkt_str = shapely.to_wkt(cell)
```

Comment:
Shouldn't we catch exception here or guard against argument not being a `Geometry`?
It's probably never going to happen, the way we use it though.

Reply:
I'm thinking about the `rounding_precision` argument for this method. It is set to `6` by default which is perfectly fine in my opinion (11 cm) but I know that people love their decimals even if it doesn't mean anything.

```python
shapely.to_wkt(shapely.from_wkt("POINT(0.12345678 0.12345678)"))
'POINT (0.123457 0.123457)'
```
by putting `rounding_precision = 8` (and `trim=False` (default) we have:

```python
shapely.to_wkt(shapely.from_wkt("POINT(0.12345678 0.12345678)"), rounding_precision=8)
'POINT (0.12345678 0.12345678)'
```
and also we don't change/extend shorter numbers:

```python
shapely.to_wkt(shapely.from_wkt("POINT(0.12345 0.12345)"), rounding_precision=8)
'POINT (0.12345 0.12345)'
```

I think we should put `rounding_precision = 8`  so we should not have complaints about us shortening the coordinates.

What do you think @joecrowleygaia ?

Reply:
Another alternative would be to set to -1 which is full precision. I'll change to 8 for now but simple enough for me to change again if needed.

Reply:
i reviewed the `value_writer` methods for the other field types both within the frictionless framework and what we have currently and i didn't find guarding of arguments provided to these methods. The type checking i did do for the `wkt_str` variable has to do with the config for shapely in mypy.ini ignoring types which sets everything to `Any` that comes out of it in the eyes of the type checker.

Reply:
ok

Reply:
pushed changes for your review @serge-gaia